pkg/chrootarchive/archive_unix.go

```diff
@@ -96,11 +96,13 @@ func invokeUnpack(decompressedArchive io.Reader, dest string, options *archive.T
 	cmd.Stderr = output
 
 	if err := cmd.Start(); err != nil {
+		w.Close()
```

Comment:
Why not defer?

Reply:
We must close the stream before `cmd.Wait()` in line 110. A defer would cause two calls to `Close()` in the non-error case.